File: cmd/pub/data/order.go
package data

import (
	"github.com/Nomoke/wb-test-app/internal/models"
)

func RandomOrder() models.Order {
	trackNumber := randomString(10)

	items := randomOrderItems(randomInt(1, 5), trackNumber)

	order := models.Order{
		OrderUID:        randomUIID(),
		TrackNumber:     trackNumber,
		Entry:           "WBIL",
		Delivery:        randomDelivery(),
		Payment:         randomPayment(),
		Items:           items,
		Locale:          "en",
		InternalSig:     randomString(19),
		CustomerID:      randomString(3),
		DeliveryService: "wb-delivery",
		ShardKey:        randomString(1),
		SMID:            randomInt(1, 100),
		DateCreated:     randomTime(),
		OofShard:        randomString(1),
	}

	return order
}

func randomDelivery() models.Delivery {
	delivery := models.Delivery{
		ID:      randomInt(1, 20),
		Name:    randomString(10),
		Phone:   randomPhone(),
		Zip:     randomString(6),
		City:    randomString(6),
		Address: randomString(10),
		Region:  randomString(11),
		Email:   randomEmail(),
	}

	return delivery
}

func randomPayment() models.Payment {
	payment := models.Payment{
		ID:           randomInt(1, 20),
		Transaction:  randomUIID(),
		RequestID:    randomString(10),
		Currency:     "RUB",
		Provider:     "wbpay",
		Amount:       randomInt(100, 10000),
		PaymentDT:    randomTime(),
		Bank:         "wb-bank",
		DeliveryCost: randomInt(700, 1500),
		GoodsTotal:   randomInt(200, 1000),
		CustomFee:    0,
	}

	return payment
}

func randomOrderItem(trackNumber string) models.OrderItem {
	item := models.OrderItem{
		ChrtId:      randomInt(100, 10000),
		TrackNumber: trackNumber,
		Price:       randomInt(1000, 5000),
		RID:         randomUIID(),
		Name:        randomString(10),
		Sale:        randomInt(0, 51),
		Size:        randomString(3),
		TotalPrice:  randomInt(250, 1600),
		NMID:        randomInt(100000, 1500000),
		Brand:       randomString(10),
		Status:      randomInt(100, 200),
	}

	return item
}

func randomOrderItems(n int, trackNumber string) []models.OrderItem {
	items := []models.OrderItem{}

	for i := 0; i < n; i++ {
		items = append(items, randomOrderItem(trackNumber))
	}

	return items
}

File: ui/lib/model/order.dart
class OrderModel {
  final String id;
  final DateTime dateCreated;
  final String trackNumber;
  final String deliveryService;
  final int deliveryId;
  final int paymentId;
  final String locale;
  final String internalSignature;
  final String customerId;
  final String shardKey;
  final int smId;
  final String oofShard;
  final Delivery delivery;
  final Payment payment;
  final List<OrderItem> items;

  OrderModel({
    required this.id,
    required this.dateCreated,
    required this.trackNumber,
    required this.deliveryService,
    required this.deliveryId,
    required this.paymentId,
    required this.locale,
    required this.internalSignature,
    required this.customerId,
    required this.shardKey,
    required this.smId,
    required this.oofShard,
    required this.delivery,
    required this.payment,
    required this.items,
  });

  factory OrderModel.fromJson(Map<String, dynamic> json) {
    return OrderModel(
      id: json['order_uid'],
      dateCreated: DateTime.parse(json['date_created']),
      trackNumber: json['track_number'],
      deliveryService: json['delivery_service'],
      deliveryId: json['delivery']['id'],
      paymentId: json['payment']['id'],
      locale: json['locale'],
      internalSignature: json['internal_signature'],
      customerId: json['customer_id'],
      shardKey: json['shard_key'],
      smId: json['sm_id'],
      oofShard: json['oof_shard'],
      delivery: Delivery.fromJson(json['delivery']),
      payment: Payment.fromJson(json['payment']),
      items: List<OrderItem>.from(
          json['items'].map((item) => OrderItem.fromJson(item))),
    );
  }
}

class Delivery {
  final int id;
  final String name;
  final String phone;
  final String zip;
  final String city;
  final String address;
  final String region;
  final String email;

  Delivery({
    required this.id,
    required this.name,
    required this.phone,
    required this.zip,
    required this.city,
    required this.address,
    required this.region,
    required this.email,
  });

  factory Delivery.fromJson(Map<String, dynamic> json) {
    return Delivery(
      id: json['id'],
      name: json['name'],
      phone: json['phone'],
      zip: json['zip'],
      city: json['city'],
      address: json['address'],
      region: json['region'],
      email: json['email'],
    );
  }
}

class Payment {
  final int id;
  final String transaction;
  final String requestId;
  final String currency;
  final String provider;
  final int amount;
  final DateTime paymentDt;
  final String bank;
  final int deliveryCost;
  final int goodsTotal;
  final int customFee;

  Payment({
    required this.id,
    required this.transaction,
    required this.requestId,
    required this.currency,
    required this.provider,
    required this.amount,
    required this.paymentDt,
    required this.bank,
    required this.deliveryCost,
    required this.goodsTotal,
    required this.customFee,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'],
      transaction: json['transaction'],
      requestId: json['request_id'],
      currency: json['currency'],
      provider: json['provider'],
      amount: json['amount'],
      paymentDt: DateTime.parse(json['payment_dt']),
      bank: json['bank'],
      deliveryCost: json['delivery_cost'],
      goodsTotal: json['goods_total'],
      customFee: json['custom_fee'],
    );
  }
}

class OrderItem {
  final int chrtId;
  final String trackNumber;
  final int price;
  final String rid;
  final String name;
  final int sale;
  final String size;
  final int totalPrice;
  final int nmId;
  final String brand;
  final int status;

  OrderItem({
    required this.chrtId,
    required this.trackNumber,
    required this.price,
    required this.rid,
    required this.name,
    required this.sale,
    required this.size,
    required this.totalPrice,
    required this.nmId,
    required this.brand,
    required this.status,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) {
    return OrderItem(
      chrtId: json['chrt_id'],
      trackNumber: json['track_number'],
      price: json['price'],
      rid: json['rid'],
      name: json['name'],
      sale: json['sale'],
      size: json['size'],
      totalPrice: json['total_price'],
      nmId: json['nm_id'],
      brand: json['brand'],
      status: json['status'],
    );
  }
}
